python/pycrdt/_map.py

```diff
@@ -108,6 +108,8 @@ def pop(self, *args) -> Any:
                     raise KeyError
                 return default_value[0]
             res = self[key]
+            if isinstance(res, BaseType):
+                res = type(res)(res.to_py())
```

Comment:
I definitely see your points, but they made me ponder for while now.

On one hand, I honestly feel uncomfortable to do a conversion on `pop()` as this is against the user's experience and expectation with ordinary Python Dicts ("I get out what I have put in"), but on the other I also cannot really assess the implications of doing a deep recursive copy.
Could you give me an example where it would be an issue to have the (false) impression that the returned shared type is the same that was put in before - but it is actually a copy?

Reply:
> Could you give me an example where it would be an issue to have the (false) impression that the returned shared type is the same that was put in before - but it is actually a copy?

For instance, all the observers that were registered on the original shared type would not work on the copy.

Reply:
Right, and I don't see an obvious way to make a copy here.
Ordinary Python objects then :+1: